### DISCUSSION TITLE
Regarding vmsg react library and Deepgarm Integration

### DISCUSSION THREAD

**Discussion by kanzariyaviral** [deepgram/community#239] [General help] [answered]
Actually I am using vmgs library for my react project. In which i am using vmgs for audio recording and  i want to translate audio into text. but didn't found any lead from anywhere there both are vmgs and deepgram are connected. I am storing my audio into URL like something that (http://localhost:3000/21d88b22-a0c0-4a96-8ff5-e5e2acf4b5a1) and i want to translate into text please help me i am waiting for your response

**Comment by lukeocodes** [comment, marked as answer]
> Actually I am using vmgs library for my react project. In which i am using vmgs for audio recording and i want to translate audio into text. but didn't found any lead from anywhere there both are vmgs and deepgram are connected. I am storing my audio into URL like something that (http://localhost:3000/21d88b22-a0c0-4a96-8ff5-e5e2acf4b5a1) and i want to translate into text please help me i am waiting for your response

Assuming at your URL returns the MP3 (or other format) you can use the URL as a URL source.

```
import { Deepgram } from "@deepgram/sdk";

const deepgram = new Deepgram(DEEPGRAM_API_KEY);

const response = await deepgram.transcription.preRecorded(
  { url: "your-vmsg-url" },
  options
);
```

It's important this returns MP3 source though, and no other information. Similarly, the URL you send must be a public URL. You can use ngrok to make your localhost URLs public, because Deepgram needs to download your audio to their server to transcribe it.

**Reply by kanzariyaviral** [reply]
@deepgram/sdk

it's a node.js module and i am using react jo when  i am integrating then getting error crypto module and many more.please add vmgs library with deepgram in react
